File: Data Structures & Algorithms/duplicate-integer/submission-0.swift
class Solution {
    func hasDuplicate(_ nums: [Int]) -> Bool {
        var map: [Int:Int] = [:]
        for num in nums {
            map[num, default:0] += 1
        }

        for (_,value) in map {
            if value >= 2 {
                return true
            }
        }
        return false
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        map = {0:1}
        sum = 0
        count = 0
        for i in range(len(nums)):
            sum += nums[i]
            leftout = sum - k
            if leftout in map:
                count += map[leftout]
            map[sum] = map.get(sum,0) + 1
        return count


        